### PR TITLE
Orders

### PR DESCRIPTION
**Pull request by shumYevh**

### CODE REVIEW

File: src/main/java/org/example/bookweb/controller/OrderController.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+@Tag(name = "Order management", description = "Endpoints for managing orders")
+@RequiredArgsConstructor
+@RestController
+@RequestMapping(value = "/orders")
+public class OrderController {
+    private final OrderService orderService;
+
+    @PostMapping
+    @Operation(summary = "",
+            description = "")
```
Comment (MAMentorFX) [inline src/main/java/org/example/bookweb/controller/OrderController.java]: add summary and description

File: src/main/java/org/example/bookweb/controller/OrderController.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+                                         @RequestBody OrderStatusUpdateRequest statusRequest) {
+        return orderService.updateStatusByOrderId(orderId, statusRequest);
+    }
+
+    @GetMapping("/{orderId}/items")
+    @Operation(summary = "",
+            description = "")
+    public Set<OrderItemResponseDto> getItemsByOrder(Authentication authentication,
+                                                     @PathVariable Long orderId) {
+        User user = (User) authentication.getPrincipal();
```
Comment (MAMentorFX) [inline src/main/java/org/example/bookweb/controller/OrderController.java]: extract to local method

**Review by MAMentorFX — CHANGES_REQUESTED** [review]

File: src/main/java/org/example/bookweb/controller/OrderController.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Order management", description = "Endpoints for managing orders")
+@RequiredArgsConstructor
+@RestController
+@RequestMapping(value = "/orders")
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookweb/controller/OrderController.java]:
```suggestion
@RequestMapping("/orders")
```

File: src/main/java/org/example/bookweb/controller/OrderController.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+                                         @PathVariable Long orderId,
+                                         @RequestBody OrderStatusUpdateRequest statusRequest) {
+        return orderService.updateStatusByOrderId(orderId, statusRequest);
+    }
+
+    @GetMapping("/{orderId}/items")
+    @Operation(summary = "Get order Items",
+            description = "Get order items from specific order and user")
+    public Set<OrderItemResponseDto> getItemsByOrder(Authentication authentication,
+                                                     @PathVariable Long orderId) {
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookweb/controller/OrderController.java]:
```suggestion
                                                     @PathVariable @Positive Long orderId) {
```
And add `@Validated` under the controller

Comment (Elena-Bruyako) [reply src/main/java/org/example/bookweb/controller/OrderController.java]: Check other places

File: src/main/java/org/example/bookweb/models/Order.java [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+                            + " for value " + strStatus);
+        }
+
+        @Override
+        public String toString() {
+            return name();
+        }
+    }
+}
+
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookweb/models/Order.java]: remove blank lines

File: src/main/java/org/example/bookweb/models/OrderItem.java [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+@Setter
+@Getter
+@Entity
+@Table(name = "order_items")
+public class OrderItem {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @ManyToOne
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookweb/models/OrderItem.java]:
```suggestion
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
```

File: src/main/java/org/example/bookweb/models/OrderItem.java [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    @JoinColumn(name = "order_id", nullable = false)
+    private Order order;
+    @ManyToOne
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookweb/models/OrderItem.java]:
```suggestion
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
```

File: src/main/java/org/example/bookweb/models/OrderItem.java [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+    @JoinColumn(name = "book_id",nullable = false)
+    private Book book;
+    @Column(nullable = false)
+    private int quantity;
+    @Column(nullable = false)
+    private BigDecimal price;
+
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookweb/models/OrderItem.java]:
```suggestion
```

File: src/main/java/org/example/bookweb/service/order/OrderServiceImpl.java [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+
+    @Override
+    public List<OrderResponseDto> getOrdersByUser(User user) {
+        return orderMapper.toDto(orderRepository.findByUserId(user.getId()));
+    }
+
+    @Override
+    public OrderResponseDto updateStatusByOrderId(Long orderId,
+                                                  OrderStatusUpdateRequest statusUpdateRequest) {
+        Order.Status statusEnum = Order.Status.fromString(statusUpdateRequest.getStatus());
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookweb/service/order/OrderServiceImpl.java]:
```suggestion
        Order order = orderRepository.findById(orderId)
                .map(o -> {
                    o.setStatus(Status.valueOf(orderDto.status()));
                    return o;
                })
                .orElseThrow(() -> new EntityNotFoundException(
                        String.format("Order with id: %d not found", orderId)
                ));
        return orderMapper.toUpdateDto(orderRepository.save(order));
```

Where 
```
@Mapping(target = "orderId", source = "id")
UpdateOrderResponseDto toUpdateDto(Order order);
```

So you don't need fromString() method

File: src/main/java/org/example/bookweb/service/order/OrderServiceImpl.java [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+        return orderItemMapper.toDto(
+                orderItemRepository.findOrderItemsByOrderIdAndUserId(orderId, user.getId())
+        );
+    }
+
+    @Override
+    public OrderItemResponseDto getOrderItemByIdAndOrderId(User user,
+                                                           Long orderId,
+                                                           Long orderItemId) {
+        return orderItemMapper.toDto(orderItemRepository.findOrderItemByIdAndOrderIdAndUserId(
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookweb/service/order/OrderServiceImpl.java]: To increase readability let's use
```suggestion
        OrderItem item = orderItemRepository.findByIdAndOrderIdAndUserId(orderItemId, orderId, userId)
                .orElseThrow(() -> new EntityNotFoundException(
                        String.format("Order item: %d not found in order: %d for user: %d",
                                orderItemId, orderId, userId)
                ));
        return orderItemMapper.toOrderItemDto(item);
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job! Let's improve your solution!

**Review by sarakhmen — APPROVED** [review]

**Pull request merged**